skmultiflow/classification/meta/adaptive_random_forests.py

```diff
@@ -0,0 +1,396 @@
+        self.instances_seen = 0
+        self._train_weight_seen_by_model = 0.0
+        self.ensemble = None
+        self.nominal_attributes = nominal_attributes
+        self._evaluator_method = ARFBaseClassifierEvaluator  # TODO use skmultiflow evaluator
+
+    def fit(self, X, y, classes=None, weight=None):
+        raise NotImplementedError
+    
+    def partial_fit(self, X, y, classes=None, weight=None):
+        if y is not None:
+            if weight is None:
+                weight = INSTANCE_WEIGHT
+            row_cnt, _ = get_dimensions(X)
+            wrow_cnt, _ = get_dimensions(weight)
+            if row_cnt != wrow_cnt:
+                weight = [weight[0]] * row_cnt                
+            for i in range(row_cnt):
+                if weight[i] != 0.0:
+                    self._train_weight_seen_by_model += weight[i]
+                    self._partial_fit(X[i], y[i], weight[i])
+        
+    def _partial_fit(self, X, y, weight):
+        self.instances_seen += 1
+        
+        if self.ensemble is None:
+            self.init_ensemble(X)
+
+        for i in range(self.nb_ensemble):
+            y_predicted = self.ensemble[i].predict(np.asarray([X]))
+            self.ensemble[i].evaluator.update(y_predicted, np.asarray([y]), weight)
+            k = np.random.poisson(self.lambda_value)
+            if k > 0:
+                self.ensemble[i].partial_fit(np.asarray([X]), np.asarray([y]), np.asarray([k]), self.instances_seen)
+    
+    def predict(self, X):
+        """Predicts the label of the X instance(s)
+        Parameters
+        ----------
+        X: numpy.ndarray of shape (n_samples, n_features)
+            Samples for which we want to predict the labels.
+        Returns
+        -------
+        list
+            Predicted labels for all instances in X.
+        """
+        r, _ = get_dimensions(X)
+        predictions = []
+        for i in range(r):
+            votes = self.get_votes_for_instance(X[i])
+            if votes == {}:
+                # Ensemble is empty, all classes equal, default to zero
+                predictions.append(0)
+            else:
+                # predict = []
+                # for vote in votes:
+                predictions.append(max(votes, key=votes.get))  # TODO Verify approach
+                # y, counts = np.unique(predict, return_counts=True)
+                # value = np.argmax(counts)
+                # predictions.append(y[value])
+        return predictions
+
+    def predict_proba(self, X):
+        raise NotImplementedError
+        
+    def reset(self):        
+        """Reset ARF."""
+        self.ensemble = None
+        self.subspace_size = 0
+        self.instances_seen = 0
+        self._train_weight_seen_by_model = 0.0
+        self._evaluator_method = ARFBaseClassifierEvaluator
+        
+    def score(self, X, y):
+        raise NotImplementedError
+        
+    def get_info(self):
+        return "NotImplementedError"
```

Comment:
Should be `raise NotImplementedError` instead of a `return "NotImplementedError"`?

Reply:
- [x] Fixed